src/test/java/com/google/research/bleth/simulator/ObserverTest.java

```diff
@@ -0,0 +1,930 @@
+        Transmission secondBeaconTransmission = secondBeacon.transmit();
+
+        randomObserver.observe(firstBeaconTransmission);
+        randomObserver.observe(secondBeaconTransmission);
+        randomObserver.passInformationToResolver();
+
+        assertThat(((FakeResolver) resolver).getObserverLocation()).isEqualTo(randomObserver.getLocation());
+    }
+
+    @Test
+    public void observerObservesTwoBeaconsPassesTheRightTransmissions() {
+        Board board = new Board(2, 2);
+        Mockito.when(simulation.getBoard()).thenReturn(board);
+        Observer randomObserver = createRandomObserverOnLocation(zeroOnZeroCoordinate);
+        Beacon firstBeacon = createStaticBeaconOnLocation(zeroOnZeroCoordinate);
+        Transmission firstBeaconTransmission = firstBeacon.transmit();
+        Beacon secondBeacon = createStaticBeaconOnLocation(zeroOnZeroCoordinate);
+        Transmission secondBeaconTransmission = secondBeacon.transmit();
+
+        randomObserver.observe(firstBeaconTransmission);
+        randomObserver.observe(secondBeaconTransmission);
+        randomObserver.passInformationToResolver();
+
+        assertThat(((FakeResolver) resolver).getTransmissions()).containsExactly(firstBeaconTransmission, secondBeaconTransmission);
+    }
+
+    @Test
+    public void transmissionsIsEmptyAfterObserverPassThem() {
+        Board board = new Board(2, 2);
+        Mockito.when(simulation.getBoard()).thenReturn(board);
+        Observer randomObserver = createRandomObserverOnLocation(zeroOnZeroCoordinate);
+        Beacon firstBeacon = createStaticBeaconOnLocation(zeroOnZeroCoordinate);
+        Transmission firstBeaconTransmission = firstBeacon.transmit();
+
+        randomObserver.observe(firstBeaconTransmission);
+        randomObserver.passInformationToResolver();
+        randomObserver.passInformationToResolver();
+
+        assertThat(((FakeResolver) resolver).getTransmissions()).isEmpty();
+    }
+
+    @Test
+    public void transmissionsRefillAfterObserverPassThem() {
+        Board board = new Board(2, 2);
+        Mockito.when(simulation.getBoard()).thenReturn(board);
+        Observer randomObserver = createRandomObserverOnLocation(zeroOnZeroCoordinate);
+        Beacon firstBeacon = createStaticBeaconOnLocation(zeroOnZeroCoordinate);
+        Transmission firstBeaconTransmission = firstBeacon.transmit();
+        Beacon secondBeacon = createStaticBeaconOnLocation(oneOnOneCoordinate);
+        Transmission secondBeaconTransmission = secondBeacon.transmit();
+
+        randomObserver.observe(firstBeaconTransmission);
+        randomObserver.passInformationToResolver();
+        randomObserver.observe(secondBeaconTransmission);
+        randomObserver.passInformationToResolver();
+
+        assertThat(((FakeResolver) resolver).getTransmissions()).containsExactly(secondBeaconTransmission);
+    }
+
+    @Test
+    public void fixedAwakenessObserverWithFirstAwakenessTimeZeroStartsAwake() {
```

Comment:
If you perform the change I had suggested on AwakenessStrategy (move its parameters inside), you could split off this set of tests and just verify the AwakenessStrategy directly.

Furthermore - you could implement the tests as an abstract test class that then applies to multiple concrete awakeness strategies (so you write the tests once, then run the tests on all implementations).